### PR TITLE
[LSP] Exclude repl output from lsp

### LINKED ISSUE #1250: Suspicion: when Calva's REPL window has a lot of text, it is slowing down clojure-lsp
As VS Code is sending all Clojure files to clojure-lsp, and the REPL/output window is a Clojure file, it is analysed and could potentially be causing performance problems affecting the whole project.

### PR DESCRIPTION
**Pull request by ericdallo**
## What has Changed?

- Never send repl content to LSP analysis 

Fixes #1250

## My Calva PR Checklist

I have:

- [X] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [X] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [X] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [X] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [X] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [ ] Added to or updated docs in this branch, if appropriate
- [X] Tests
     - [X] Tested the particular change
     - [X] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
     - [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test.
- [X] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [X] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [ ] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [X] Created the issue I am fixing/addressing, if it was not present.

Ping @pez, @bpringe

### CODE REVIEW

File: src/lsp/main.ts [hunk 1]
```diff
@@ -64,6 +64,24 @@ function createClient(clojureLspPath: string): LanguageClient {
...
+                return next(document);
+            },
+            didSave: async (document, next) => {
+                if (document.uri.path.endsWith(config.REPL_FILE_EXT)) {
+                    return;
+                }
+                return next(document);
+            },
+            didChange: async (change, next) => {
+                if (change.document.uri.path.endsWith(config.REPL_FILE_EXT)) {
```
Comment (PEZ) [inline src/lsp/main.ts]: We should probably use `isResultsDoc(doc)` from `results-doc.ts` here.

I noticed that there was prior art in this file doing it this way, so we should fix that one too.

A reason to check the extension is if we consider that Calva would have more than one output/REPL window, but if that happens we will need to revisit a lot of code, niot just this one.

**Comment by PEZ** [comment]
Thanks!

**Comment by PEZ** [comment]
There seems to be some unrelated commits in the PR branch. Can you clean that up? (I'm not sure this is necessary, but it is generally easier to deal with ”clean” PRs :smile:.)

**Comment by PEZ** [comment]
@bpringe Are we relying on clojure-lsp in a way that we need to consider before merging this? I seem to recall something about the debugger using clojure-lsp analysis.

File: src/lsp/main.ts (hunk 1 above)
Comment (ericdallo) [reply src/lsp/main.ts]: Done, I found no way to fix the one that was already being used on this file, we would need to convert a vscode.Uri to a vscode.TextDocument

**Comment by ericdallo** [comment]
Thanks @PEZ ! Done, LMK if any more issues

File: src/lsp/main.ts (hunk 1 above)
Comment (PEZ) [reply src/lsp/main.ts]: I see. Might be the reason it was not done like that. 😄

**Comment by bpringe** [comment]
@PEZ The debugging code just uses clojure-lsp for decorations (for the instrumented functions). It doesn't seem to even add decorations in the repl window currently, and I'm unsure if that's intentional, but either way, I don't think it matters if this disallows that.

**Comment by bpringe** [comment]
Thanks @ericdallo! Let's merge. 🚀

**Comment by bpringe** [comment]
By the way, I removed that newline between changes in the unreleased section of the changelog (on dev). It might have caused an issue with our publish script.

**Pull request merged**